src/DataSource.ts

```diff
@@ -70,8 +70,10 @@ export class DataSource extends DataSourceApi<MyQuery, MyDataSourceOptions> {
   private createQuery(query: MyQuery, range: TimeRange | undefined, scopedVars: ScopedVars | undefined = undefined) {
     let payload = query.queryText;
     if (range) {
-      payload = payload.replace(/\$timeFrom/g, range.from.valueOf().toString());
-      payload = payload.replace(/\$timeTo/g, range.to.valueOf().toString());
+      payload = getTemplateSrv().replace(payload, {
+        timeFrom: { text: 'from', value: range.from.valueOf() },
```

Comment:
What do the `text` values mean? Does it matter what they're set to?

Reply:
To be honest, I don't know what the `text` property is and what it is used for. It is required by `ScopedVar` interface, though (https://github.com/grafana/grafana/blob/7520ebadacf2f5276a1f187f1bf4cfc4b953ab23/packages/grafana-data/src/types/ScopedVars.ts#L1). I didn't spend too much time selecting a good value for that property, but I can change it for a more meaningful message. Any suggestions?.

Reply:
As for the `date` format options, they worked just fine for me with `timeFrom` and `timeTo`:

![Screen Shot 2020-10-30 at 8 56 09 PM](https://user-images.githubusercontent.com/64748/97770163-8c1e3900-1af6-11eb-821b-905b859def80.png)

What version of Grafana did you use for your tests?. I think the `date` format options were introduced in Grafana v7.2: https://grafana.com/blog/2020/09/23/grafana-v7.2-released-with-custom-date-formats-new-transforms-and-overrides/

Reply:
>  I can change it for a more meaningful message

Nah, it's fine, I was having trouble figuring out what it was used for myself so just was curious if you knew. :)

> As for the date format options, they worked just fine for me with timeFrom and timeTo

You're right, sorry. It seems while I was running 7.2.2 backend I was still getting some old frontend code.